main.go

```diff
@@ -20,9 +23,35 @@ func main() {
 		log.Fatalf("Failed to parse %s: %v", *yml, err)
 	}
 
+	var templates *template.Template
+	if *tpls != "" {
+		log.Printf("Parsing templates at path: %s\n", *tpls)
+		templates, err = getCombinedTemplates(*tpls)
+		if err != nil {
+			log.Fatalf("Failed to parse templates at %s: %v", *tpls, err)
+		}
+	} else {
+		templates = _templates
+	}
+
 	log.Printf("Creating HTTP handler with config: %v", config)
-	handler := CreateHandler(config)
+	handler, err := CreateHandler(config, templates)
+	if err != nil {
+		log.Fatalf("Failed to create handler: %v", err)
+	}
 
 	log.Printf(`Starting HTTP handler on ":%d"`, *port)
 	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), handler))
 }
+
+func getCombinedTemplates(dir string) (*template.Template, error) {
+	// Clones default templates to then merge with the user defined templates.
+	// This allows for the user to only override certain templates, but not all
+	// if they don't want.
+	templates, err := _templates.Clone()
+	if err != nil {
+		return nil, err
+	}
+	templates, err = templates.ParseGlob(filepath.Join(dir, "*.html"))
+	return templates, err
```

Comment:
optional: can shorten to

```suggestion
	return templates.ParseGlob(filepath.Join(dir, "*.html"))
```